os/net/netdev/netdev_ioctl.c

```diff
@@ -1067,6 +1066,42 @@ int lwip_func_ioctl(FAR struct socket *sock, int cmd, void  *arg)
 
 	return ret;
 }
+
+/****************************************************************************
+ * Function: lwipioctl
+ *
+ * Description:
+ *   Call lwip_ioctl() with FIONREAD/FIONBIO commands or
+ *   call lwip API with SIOCLWIP command
+ *
+ * Parameters:
+ *   sockfd   Socket file descriptor
+ *   cmd      The ioctl command
+ *   arg      Type of the information to get
+ *
+ * Returned Value:
+ *   0 on success, negated errno on failure.
+ *
+ ****************************************************************************/
+
+int lwipioctl(int sockfd, int cmd, void *arg)
+{
+	int ret = -ENOTTY;
+
+	if (cmd == FIONREAD || cmd == FIONBIO) {
+		ret = lwip_ioctl(sockfd, (long)cmd, arg);
+		if (ret == 0) {
```

Comment:
As I left before, this line is not necessary.
```
ret = lwip_ioctl(...);
if (ret == -1) {
   ret = -get_errno();
}
```
This is enough.

Reply:
updated